spec/requests/doorkeeper/tokens_controller_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Doorkeeper::TokensController, type: :request do
+  context 'when user ask for a token with good request' do
+    it 'return a refreshable token' do
+      post 'create', params: token_request_params
+      expect(JSON.parse(response.body)['token_type']).to eq('Bearer')
+    end
+  end
+
+  context 'when user ask for a token without client secret' do
+    it 'return invalid_client error' do
+      params = token_request_params
+      params.except!(:client_secret)
+
+      post 'create', params: params
+
+      expect(JSON.parse(response.body)['error']).to eq('invalid_client')
+    end
+  end
+
+  context 'when user ask for a token with bad credentials' do
+    it 'return invalid_grant error' do
+      params = token_request_params
```

Comment:
can we simplify it in one line

```
  params = token_request_params.merge(password: 'wrong_pass')
```

Reply:
the beauty of mastering ruby!! :) 
Nice!! :) 👍

Reply:
Applied here too:
![image](https://user-images.githubusercontent.com/77609814/122364189-c05e4100-cf83-11eb-9987-adb847181224.png)